recipe/build-mpi.sh

```diff
@@ -75,27 +82,30 @@ fi
             --with-mpi-moduledir='${includedir}' \
             --with-wrapper-ldflags="${wrapper_ldflags}" \
             --with-sge \
+            --with-pmix=internal \
             --with-hwloc=$PREFIX \
             --with-libevent=$PREFIX \
             --with-zlib=$PREFIX \
-            --enable-mca-dso \
```

Comment:
I think we need to make sure this option is still enabled, otherwise optional feature support becomes required at runtime

Reply:
Restored it now